signac/common/config.py

```diff
@@ -20,51 +20,33 @@ def _get_project_config_fn(root):
     return os.path.abspath(os.path.join(root, PROJECT_CONFIG_FN))
 
 
-def _get_config_dirname(fn):
-    # We could use pathlib's `.paths` attribute to remove the trailing project
-    # config filename, but that has significant performance
-    # implications: this takes ~200 ns, whereas
-    # os.path.join(*(pathlib.PosixPath(fn).parts[:-2])) takes ~6 us, almost a
-    # 20x slowdown.
-    return fn.replace(os.sep + PROJECT_CONFIG_FN, "")
-
-
-def _search_local(root):
-    fn_ = _get_project_config_fn(root)
-    if os.path.isfile(fn_):
-        yield fn_
-
-
-def _search_tree(root=None):
-    """Locates signac configuration files in a directory hierarchy.
+def _locate_config_dir(root):
```

Comment:
A better name might be `search_path`. The _returned_ value is the "root" of the project, not the argument here.
```suggestion
def _locate_config_dir(search_path):
```